core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala

```diff
@@ -61,6 +61,9 @@ private[spark] trait TaskScheduler {
    */
   def killTaskAttempt(taskId: Long, interruptThread: Boolean, reason: String): Boolean
 
+  // Kill all the running task attempts in a stage.
+  def killAllTaskAttempts(stageId: Int, interruptThread: Boolean, reason: String): Unit
```

Comment:
why do we need this?

Reply:
I'm also confused here. Is it part of this PR? We should kill all task attempts in case of any task failures in a barrier stage, not limited to context.barrier() failures. Right?

Reply:
Submitted #21943 , however, still need this here until #21943  is merged, because otherwise the test cases shall fail.

Reply:
IIRC killing all tasks is just the best effort, we can guarantee the tasks are all killed. Shall we tolerate this in the barrier scheduling?